pkg/network/usm/monitor.go

```diff
@@ -437,3 +487,44 @@ func (m *Monitor) createStaticTable(mgr *ebpfProgram) error {
 	}
 	return nil
 }
+
+// initProtocols takes the network configuration `c` and uses it to initialise
+// the enabled protocols' monitoring, and configures the ebpf-manager `mgr`
+// accordingly.
+//
+// For each enabled protocols, a protocol-specific instance of the Protocol
+// interface is initialised, and the required maps and tail calls routers are setup
+// in the manager.
+//
+// If a protocol is not enabled, its tail calls are instead added to the list of
+// excluded functions for them to be patched out by ebpf-manager on startup.
+//
+// It returns:
+// - a slice containing instances of the Protocol interface for each enabled protocol support
+// - a slice containing pointers to the protocol specs of disabled protocols.
+// - an error value, which is non-nil if an error occurred while initialising a protocol
+func initProtocols(c *config.Config, mgr *ebpfProgram) (map[protocols.ProtocolType]protocols.Protocol, []*protocols.ProtocolSpec, error) {
+	enabledProtocols := make(map[protocols.ProtocolType]protocols.Protocol)
+	disabledProtocols := make([]*protocols.ProtocolSpec, 0)
+
+	for proto, spec := range knownProtocols {
+		protocol, err := spec.Factory(c)
+		if err != nil {
+			return nil, nil, &errNotSupported{err}
```

Comment:
I now see that if one protocol isn't supported, we won't initialize the monitor at all. Why is that?

Reply:
The rationale is, if you end up in this case it means the configuration asked to enable a protocol while its not supported. That, to me, would be a configuration error which needs to be solved by the user hence I why I made it fail early. Its also the behaviour we had before the refactor.